src/array_slice.cpp

```diff
@@ -732,7 +732,13 @@ std::vector<double> fields::get_array_metadata(const volume &where, bool collaps
     full_dirs[fr] = dirs[fr];
 
   double *weights = get_array_slice(where, NO_COMPONENT);
-  if (collapse_empty_dimensions) weights = collapse_array(weights, &rank, dims, dirs, where);
+  if (collapse_empty_dimensions) {
+    double *newweights = collapse_array(weights, &rank, dims, dirs, where);
+    if (weights != newweights) {
```

Comment:
This line *always* evaluates to false because in the line just before this one, `weights` is passed by reference to `collapse_array` which deletes it at the end:

https://github.com/NanoComp/meep/blob/8af7808bf674481a71a1c04a2fc27c1a624e5e4e/src/array_slice.cpp#L708-L710

Thus, the only thing that is necessary to fix the memory leak is to include the line `delete[] newweights;` replacing `delete[] weights;` in this PR.

Reply:
> This line always evaluates to false because in the line just before this one, weights is passed by reference to collapse_array which deletes it at the end:

Actually, it only deletes it if `collapse_array` creates a new array. There are certain cases where it simply returns the existing `weights` array, which is why it's important to check if it's the same as before.

I think this is correct.

Reply:
I modified my original comment to point out that the `delete[] weights;` line inside the if statement is actually unnecessary since whenever the `weights` array *is* collapsed, it is deleted inside `collapse_array`.

Reply:
Right, I missed that `collapse_array` was already deleting the old array in the case where it allocates a new one.  Should be fixed.